tests/test_autosave.py

```diff
@@ -0,0 +1,418 @@
+from conftest import get_multiprocessing_context, select_and_recv
+from softioc import autosave, builder, softioc
+from unittest.mock import patch
+import pytest
+import threading
+import numpy
+import re
+import yaml
+import time
+
+DEVICE_NAME = "MY-DEVICE"
+
+
+@pytest.fixture(autouse=True)
+def reset_autosave_setup_teardown():
+    default_pvs = autosave.Autosave._pvs.copy()
+    default_state = autosave.Autosave._last_saved_state.copy()
+    default_save_period = autosave.Autosave.save_period
+    default_device_name = autosave.Autosave.device_name
+    default_directory = autosave.Autosave.directory
+    default_enabled = autosave.Autosave.enabled
+    default_tb = autosave.Autosave.timestamped_backups
+    yield
+    autosave.Autosave._pvs = default_pvs
+    autosave.Autosave._last_saved_state = default_state
+    autosave.Autosave._stop_event = threading.Event()
+    autosave.Autosave.save_period = default_save_period
+    autosave.Autosave.device_name = default_device_name
+    autosave.Autosave.directory = default_directory
+    autosave.Autosave.enabled = default_enabled
+    autosave.Autosave.timestamped_backups = default_tb
+
+
+@pytest.fixture
+def existing_autosave_dir(tmp_path):
+    state = {
```

Comment:
As this data is used in several tests, I'd be tempted to split it out into its own top level dictionary or a fixture, to make it obvious that these values will be references in many tests. Not required, just a nice to have.